calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -0,0 +1,324 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from StringIO import StringIO
+from mock import patch, Mock, call
+from netaddr import IPAddress
+from calico_ctl.bgp import *
+from calico_ctl.bgp import validate_arguments as bgp_validate_arguments
+from calico_ctl.endpoint import validate_arguments as ep_validate_arguments
+from calico_ctl.node import validate_arguments as node_validate_arguments
+from calico_ctl.pool import *
+from calico_ctl.pool import validate_arguments as pool_validate_arguments
+from calico_ctl.profile import validate_arguments as profile_validate_arguments
+from calico_ctl.container import validate_arguments as container_validate_arguments
+from pycalico.datastore_datatypes import BGPPeer
+
+
+class TestBgp(unittest.TestCase):
+
+    @patch('calico_ctl.bgp.sys.exit',autospec=True)
+    def test_validate_arguments(self, m_sys_exit):
+        """
+        Test validate_arguments for calicoctl bgp command
+        """
+        # Set up arguments
+        cases = (({'<PEER_IP>':'127.a.0.1'},False),
+                 ({'<PEER_IP>':'aa:bb::zz'},False),
+                 ({'<AS_NUM>':9},True),
+                 ({'<AS_NUM>':'9'},True),
+                 ({'<AS_NUM>':'nine'},False),
+                 ({'show':1,'--ipv4':1},True))
+
+        # Call method under test for each test case
+        # Assert that method exits on bad input
+        for case, is_valid in cases:
+            print "Testing case %s ..." % case
+            bgp_validate_arguments(case)
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+    @patch('calico_ctl.bgp.check_ip_version',autospec=True)
+    @patch('calico_ctl.bgp.BGPPeer',autospec=True)
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_bgp_peer_add(self,m_client,m_BGPPeer,m_check_ipv):
+        """
+        Test bgp_peer_add function for calico_ctl bgp
+        """
+        # Set up mock objects
+        address = '1.2.3.4'
+        peer =  Mock(spec=BGPPeer)
+        m_check_ipv.return_value = address
+        m_BGPPeer.return_value = peer
+
+        # Call method under test
+        bgp_peer_add('1.2.3.4','v4',1)
+
+        # Assert
+        m_check_ipv.assert_called_once_with('1.2.3.4','v4',IPAddress)
+        m_BGPPeer.assert_called_once_with(address,1)
+        m_client.add_bgp_peer.assert_called_once_with('v4',peer)
```

Comment:
Seems like there are several places where you call functions without spaces between arguments in this function and in others.  I'll make a couple of notes for other functions I see that are like this.